torch/csrc/distributed/c10d/FlightRecorder.cpp

```diff
@@ -316,51 +300,14 @@ std::optional<FlightRecorder::Entry> FlightRecorder::getEntry(
 
 void FlightRecorder::retire_id(
     std::optional<size_t> id,
-    bool compute_duration) {
-  if (!enabled_ || !id) {
+    std::optional<float> duration) {
+  auto entry = getEntry(id);
+  if (!entry)
     return;
```

Comment:
Is there a potential race here - assuming multiple threads are working on the same entry?
For example, we do `getEntry` and it returns slot `0` entry.
But, before we get line `308`, that slot gets overwritten by the `record` function?

Reply:
Good point. I should move the `lock_guard` in line 307 to the top. Thanks.

Reply:
I think will deadlock if you just move it up - since `getEntry` also attempts to grab the same mutex_. Maybe need some refactoring to pass the lock? Or remove lock from getEntry and assume that it can only be called with a lock held?

Reply:
Yeah, the lock in getEntry should be removed. Otherwise we will need to use a recursive mutex.